src/main/kotlin/textcalculator/text/InputString.kt

```diff
@@ -0,0 +1,49 @@
+package textcalculator.text
+
+class InputString(
+    originalInput: String,
+    initialDelimiters: List<String> = listOf(",", ":")
+) {
+    private val input: String
+    private val delimiters: List<String>
+
+    init {
+        val (customDelimiter, customInput) = extractCustomInput(originalInput)
+
+        this.delimiters = determineDelimiters(initialDelimiters, customDelimiter)
+        this.input = determineInput(customDelimiter, customInput, originalInput)
+    }
+
+    private fun determineDelimiters(initialDelimiters: List<String>, customDelimiter: String): List<String> {
+        return if (customDelimiter.isNotEmpty()) initialDelimiters + customDelimiter else initialDelimiters
+    }
+
+    private fun determineInput(customDelimiter: String, customInput: String, originalInput: String): String {
+        val inputToValidate = if (customDelimiter.isNotEmpty()) customInput else originalInput
+        validate(inputToValidate, customDelimiter)
+        return inputToValidate
+    }
+
+    private fun extractCustomInput(input: String): Pair<String, String> {
+        return CUSTOM_DELIMITER_CHECK.find(input)?.let {
+            it.groupValues[1] to it.groupValues[2]
+        } ?: ("" to input)
+    }
```

Comment:
p3; 코틀린 구조 분해 선언을 활용한다면 인덱스 번호를 지정하지 않아도 될 거 같아요. 😊

```suggestion
private fun extractCustomInput(input: String): List<String> = CUSTOM_DELIMITER_REGEX.find(input = input)
        ?.let(this::splitCustomDelimiterToken)
        ?: text.split(regex = CONST) ex -> """[,:]"""

    private fun splitCustomDelimiterToken(matchResult: MatchResult): List<String> {
        val (customDelimiter, tokenStrings) = matchResult.destructured
        return tokenStrings.split(customDelimiter)
    }
```

Reply:
말씀주신대로 matchResult의 구조분해를 이용해서 인덱스 없이 하도록 변경했습니당! 🤩 예시 구문에는 `extractCustomInput` 에 `List<String>` 으로 tokenStrings을 반환하도록 제시해주셨는데, 로직 상 Pair로 받아서 customDelimiter & tokenStrings 모두 필요해서 리턴은 Pair로 그대로 하도록 했슴니다! 😊👍

Reply:
Pair의 사용처를 없애기 위함과 최대한 메서드들을 퓨어 메서드로 만들도록 유도한 거기도 합니다. 😊
현재 미션에서 중요한 내용은 아니니 참고만 해 주시면 될 거 같아요. 👍